psa_builder.py

```diff
@@ -237,7 +236,32 @@ def check_and_clone_repo(name, deps, dir):
         logging.info(
             "%s repo exists, fetching latest from remote %s", name, deps
         )
-        cmd = ["git", "-C", join(dir, name), "fetch", deps]
+        cmd = [
+            "git",
+            "-C",
+            tfm_dir,
+            "remote",
+            "get-url",
+            deps,
+        ]
+        ret = run_cmd_and_return(cmd)
+        if ret != 0:
+            logging.info("%s is not a remote, adding it", deps)
+            cmd = [
+                "git",
+                "-C",
+                tfm_dir,
+                "remote",
+                "add",
+                deps,
+                dependencies[deps].get(name)[0],
+            ]
+            ret = run_cmd_and_return(cmd)
+            if ret != 0:
+                logging.critical("Failed to add remote %s", deps)
+                sys.exit(1)
```

Comment:
This is kind of preexisting as you're just following existing patterns... Shouldn't we just wrap `_main` with a `sys.exit` and come up with a proper exception handling strategy? If I ever want to call this from some other code it could possibly exit the interpreter without giving me any chance to handle the error, which isn't very considerate.

Reply:
Filed https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/issues/104 to address the whole codebase